src/main/java/org/semux/core/TransactionExecutor.java

```diff
@@ -98,104 +98,96 @@ public TransactionExecutor(Config config) {
 
             // check nonce
             if (nonce != acc.getNonce()) {
-                result.setError(Error.INVALID_NONCE);
+                result.setCode(Code.INVALID_NONCE);
                 continue;
             }
 
             // check fee
             if (fee.lt(config.minTransactionFee())) {
-                result.setError(Error.INVALID_FEE);
+                result.setCode(Code.INVALID_FEE);
                 continue;
             }
 
             // check data length
             if (data.length > config.maxTransactionDataSize(type)) {
-                result.setError(Error.INVALID_DATA_LENGTH);
+                result.setCode(Code.INVALID_DATA);
                 continue;
             }
 
             switch (type) {
             case TRANSFER: {
                 if (fee.lte(available) && value.lte(available) && sum(value, fee).lte(available)) {
-
                     as.adjustAvailable(from, neg(sum(value, fee)));
                     as.adjustAvailable(to, value);
-
-                    result.setSuccess(true);
                 } else {
-                    result.setError(Error.INSUFFICIENT_AVAILABLE);
+                    result.setCode(Code.INSUFFICIENT_AVAILABLE);
                 }
                 break;
             }
             case DELEGATE: {
                 if (!validateDelegateName(data)) {
-                    result.setError(Error.INVALID_DELEGATE_NAME);
+                    result.setCode(Code.INVALID_DELEGATE_NAME);
                     break;
                 }
                 if (value.lt(config.minDelegateBurnAmount())) {
-                    result.setError(Error.INVALID_DELEGATE_BURN_AMOUNT);
+                    result.setCode(Code.INVALID_BURNING_AMOUNT);
```

Comment:
I think I prefer keeping Delegate in the code variable name, since the act they're doing is registering a delegate, the burning is just a side effect.